targets/TARGET_NXP/TARGET_LPC176X/TARGET_UBLOX_C027/ublox_low_level_api.c

```diff
@@ -42,6 +42,7 @@ void ublox_mdm_init(void)
     // Can't use wait_ms() as RTOS isn't initialised yet
```

Comment:
Hadn't spotted that this was a requirement of the new ticker API, had kinda assumed that any changes were backwards compatible.  Obviously not.  Are there any other things of this nature that we have missed?  Note that, for reasons I've never understood, you don't run C027 on your test farms.

Reply:
The old API is not longer there, we don't maintain two versions at the same time. The calls are for most part the same, just some additional requirements are in place.

Reply:
Looking at this closer, ```us_ticker_read``` seems to be a device hal API where the platform API seems to be ```ticker_read_us``` which seems to start the ticker if not already started. ```wait_us``` still yields to the RTOS so not a good option for where this needing to be called from.

Reply:
I think I see at least one of my sources of confusion: C027 defines the macro `TARGET_LPC1768` but doesn't actually inherit the target `LPC1768`, just the target `LPCTarget`, which doesn't do much of interest; hence the C027 target needs to add any `"device_has"` macros all by it's little 'ole self (I'm used to the STM targets which are much more hierarchical).  We will add support for `USTICKER`.

Now to my other source of confusion: since there is a `ticker_api` and a `us_ticker_api` in HAL, both of which serve our purpose for this case (since we can read microseconds from them), which would be the correct one to use in this "mid-boot" state, or doesn't it matter?

Reply:
ticker_api is just a "parent" to us_ticker_api and lp_ticker_api. The whole timers/tickers abstraction is a bit complicated, but basically it's your standard object oriented abstract/virtual inheritance. You can use all of the high level classes with any ticker that inherits from ticker_api. So to instantiate us ticker in terms of ticker_api you use `get_us_ticker_data` and then interact with that using ticker_api.